processor/memorylimiter/memorylimiter.go

```diff
@@ -219,3 +242,26 @@ func (ml *memoryLimiter) memLimiting(ms *runtime.MemStats) {
 		runtime.GC()
 	}
 }
+
+type dropDecision struct {
+	memAllocLimit uint64
+	memSpikeLimit uint64
+}
+
+func (d dropDecision) shouldDrop(ms *runtime.MemStats) bool {
+	return d.memAllocLimit <= ms.Alloc || d.memAllocLimit-ms.Alloc <= d.memSpikeLimit
+}
+
+func newFixedDecision(memAllocLimit, memSpikeLimit uint64) (*dropDecision, error) {
+	if memSpikeLimit >= memAllocLimit {
+		return nil, errMemSpikeLimitOutOfRange
+	}
+	return &dropDecision{
+		memAllocLimit: memAllocLimit,
+		memSpikeLimit: memSpikeLimit,
+	}, nil
+}
+
+func newPercentrageDecision(totalMemory, percentageLimit, percentageSpike int64) (*dropDecision, error) {
+	return newFixedDecision(uint64(percentageLimit*totalMemory)/100, uint64(percentageSpike*totalMemory)/100)
```

Comment:
It will be good to limit the configured value to 100 percent in order to avoid unexpected behavior (possible OOM) due to typos (eg.: "755" instead of "75").